From dfcf8aa426d89c77b9b9594409efb70b344e02ec Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Fri, 25 Sep 2020 02:11:23 -0700
Subject: [PATCH 1680/1851] drm: xlnx: zynqmp_disp: Fix MIXED_ENUM_TYPE type
 warnings

commit 853e8b91335576842eaa80fdb89416f522c858fd from
https://github.com/Xilinx/linux-xlnx.git

This patch fix MIXED_ENUM_TYPE type coverity warnings.

Event mixed_enum_type: enumerated type mixed with another type.

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 73825394ca68..387883f04f09 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -1897,7 +1897,7 @@ static int zynqmp_disp_layer_enable(struct zynqmp_disp *disp,
 {
 	struct device *dev = disp->dev;
 	struct dma_async_tx_descriptor *desc;
-	enum dma_ctrl_flags flags;
+	unsigned long flags;
 	unsigned int i;
 
 	if (layer->enabled && layer->mode != mode) {
@@ -2097,7 +2097,7 @@ static int zynqmp_disp_layer_create(struct zynqmp_disp *disp)
 		char temp[16];
 
 		layer = &disp->layers[i];
-		layer->id = i;
+		layer->id = (enum zynqmp_disp_layer_type)i;
 		layer->offset = i * 4;
 		layer->other = &disp->layers[!i];
 		layer->num_chan = num_chans[i];
-- 
2.31.1

